include/circt/Dialect/FIRRTL/FIRRTLTypes.h

```diff
@@ -129,11 +129,12 @@ class FIRRTLBaseType : public FIRRTLType {
   /// Get the sub-type of a type for a field ID, and the subfield's ID. Strip
   /// off a single layer of this type and return the sub-type and a field ID
   /// targeting the same field, but rebased on the sub-type.
-  std::pair<FIRRTLBaseType, unsigned> getSubTypeByFieldID(unsigned fieldID);
+  std::pair<circt::hw::FieldIDTypeInterface, unsigned>
+  getSubTypeByFieldID(unsigned fieldID);
 
   /// Return the final type targeted by this field ID by recursively walking all
   /// nested aggregate types. This is the identity function for ground types.
-  FIRRTLBaseType getFinalTypeByFieldID(unsigned fieldID);
+  circt::hw::FieldIDTypeInterface getFinalTypeByFieldID(unsigned fieldID);
```

Comment:
Hmm, took me a sec to sort out that `FieldIDTypeInterface` was .. a type.  Naming it `FieldIDType` seems friendlier, and I'll note MLIR ends up generating things like `FieldIDTypeInterfaceInterfaceTraits`.

Looking around, we don't seem to be consistent here-- sometimes we use the `Like` suffix (which I think has its own meaning re:grouping behavior over similar ops, less so for any sort of extension points), sometimes they're the name without any suffix, and plenty also have the Interface suffix.

Resolving is out of scope for this PR for sure.

Looks like MLIR is pretty set on the "Interface" suffix so maybe I'll just get over it and adapt :wink: .